tardis/utilities/utils.py

```diff
@@ -131,3 +131,10 @@ def submit_cmd_option_formatter(options: AttributeDict) -> str:
             option_string += tmp_option_string
 
     return option_string.strip()
+
+
+def convert_to(value: Any, convert_to_type: Type, default: Any = None):
+    try:
+        return convert_to_type(value)
+    except ValueError:
+        return default
```

Comment:
Should we actually go with a default ``None`` in any case here? I imagine this could hide some errors.

A sentinel could be used here to differentiate between "no default" and "default of ``None``".
```suggestion
sentinel = object()

def convert_to(value: Any, convert_to_type: Type, default: Any = sentinel):
    try:
        return convert_to_type(value)
    except ValueError:
        if default is sentinel:
            raise
        return default
```

Reply:
Alternatively with naive typing:
```suggestion
from typing import TypeVar, Callable
T = TypeVar("T")
sentinel = object()

def convert_to(value: Any, convert_to_type: Callable[[Any], T], default: Any = sentinel) -> T:
    try:
        return convert_to_type(value)
    except ValueError:
        return default
```
The ``Callable`` is a bit more general than just types. Handling the ``default`` properly needs some boilerplate ``@overload``s, though, not sure if we want those.